src/core/pal/pointset.cpp

```diff
@@ -701,8 +690,14 @@ CHullBox PointSet::compute_chull_bbox()
       bbox[3] = y[cHull[i]];
   }
 
+  OrientedConvexHullBoundingBox finalBb;
 
-  dref = bbox[2] - bbox[0];
+  const double dref = bbox[2] - bbox[0];
+  if ( qgsDoubleNear( dref, 0 ) )
+  {
+    ok = false;
+    return finalBb;
+  }
 
```

Comment:
It seems to me that it makes sense to apply such fixes to all the PAL code.
For example here:

https://github.com/qgis/QGIS/blob/ffbedde2cfe445cafdaf5520a255fdaff3efad85/src/core/pal/pointset.cpp#L598-L599

or do something like this:

https://github.com/qgis/QGIS/blob/ffbedde2cfe445cafdaf5520a255fdaff3efad85/src/core/pal/pointset.cpp#L440

Reply:
Agreed, but the pal code is littered with stuff like this, and I'm not comfortable backporting large refactoring changes to the stable branches.